### ISSUE TITLE
Unable to convert ONNX

### ISSUE THREAD

**Issue by ZTMIDGO** [josephrocca/rwkv-v4-web#3]
Hi, I tried using https://colab.research.google.com/github/josephrocca/rwkv-v4-web/blob/main/RWKV_v4_ONNX_conversion.ipynb to convert to onnx model, model address  https://huggingface.co/BlinkDL/rwkv-4-pile-1b5/blob/main/RWKV-4-Pile-1B5-EngChn-testNovel-301-ctx2048-20230214.pth , but hint torch.onnx.CheckerError: The model does not have an ir_ Version set properly conversion error

**Comment by josephrocca** [comment]
Yeah, I haven't really looked into it properly yet, but I was going to ask @AXKuhta and @BlinkDL whether they knew what was going on there - maybe there's something that's architecturally slightly different with models larger than 430m. None of the 1B5 models seem to work - they throw that error during conversion.

Yesterday I got a non-quantized version of the 430m model working in wasm with the .ort format (which [doesn't needlessly replicate the params in memory](https://github.com/microsoft/onnxruntime/issues/13445#issuecomment-1430153341) like the .onnx format), and that's 1.73 GB (it's not quantized), and runs at ~12 tokens/sec - see `rwkv-4-pile-430m.with_runtime_opt.ort` in web demo. So it seems plausible that a uin8 quantized version of 1B5 would be able to run pretty easy on the wasm backend - maybe even the 3B model too!

(Note though that uint8 [seems to slow ONNX Runtime down quite a bit](https://github.com/microsoft/onnxruntime/issues/6732#issuecomment-1431238975) for some reason. Seems like such a significantly slowdown that it must be some sort of bug.)

**Comment by josephrocca** [comment]
Actually, thinking about this a bit more, I bet this is related to .onnx format's 2GB limit, since it uses protobuf.

https://github.com/microsoft/onnxruntime/issues?q=is%3Aissue+is%3Aopen+2gb+protobuf

Pretty sure this is the issue. When model files are >2GB, large tensors are stored in separate protobuf files and then they're all zipped up together. Pytorch export probably isn't handling this >2GB case properly.

Ideally there'd be some way to get it to just go straight to the .ort format, which doesn't have that limit, since even if we manage to get onnx export working, [`convert_onnx_models_to_ort` doesn't currently work on the external tensors zip format](https://github.com/microsoft/onnxruntime/issues/14697).

**Comment by ZTMIDGO** [comment]
I tried turning 1B5 to uint8, code https://github.com/harrisonvanderbyl/rwkvstic seems to be provided here, and then tried to convert to onnx, hoping this would work.Thank you

**Comment by ZTMIDGO** [comment]
Hi,You can now convert RWKV to ONNX.

https://github.com/harrisonvanderbyl/rwkvstic#onnx-export

**Comment by josephrocca** [comment]
Nice! Have you tried it? Would be cool if we could get the 1.5B model working in the browser. We very likely need a uint8 version due to wasm memory limit (currently 4GB until: https://github.com/WebAssembly/memory64), but a non-quantized version may still work with the WebGL backend.

**Comment by ZTMIDGO** [comment]
I tried, it didn't work, no way to run on the mobile platform onnxruntime, loading onnx model failed.

**Comment by ZTMIDGO** [comment]
I tried running the exported onnx model using the onnx runtime of the mobile platform, but it doesn't work, but if I export onnx using https://github.com/AXKuhta/RWKV-LM/tree/onnx, it can run on the mobile platform, unfortunately it is not possible to export the model above 1B5.

**Comment by harrisonvanderbyl** [comment]
possible issues:
new rwkvsic onnx export has external data: .bin file,
new rwkvstic onnx export inputs/outputs are flattened
